target/linux/lantiq/files/arch/mips/boot/dts/lantiq/vr9.dtsi

```diff
@@ -446,22 +430,71 @@
 			};
 		};
 
-		eth0: eth@e108000 {
+		gswip: switch@e108000 {
+			compatible = "lantiq,xrx200-gswip";
 			#address-cells = <1>;
 			#size-cells = <0>;
+			reg = < 0xe108000 0x3000 /* switch */
+				0xe10b100 0x70   /* mdio */
+				0xe10b1d8 0x30   /* mii */
+				>;
+
+			dsa,member = <0 0>;
+
+			gswip_ports: ports {
+				#address-cells = <1>;
+				#size-cells = <0>;
+
+				port@6 {
+					reg = <0x6>;
+					label = "cpu";
+					ethernet = <&eth0>;
+				};
+			};
+
+			gswip_mdio: mdio {
+				#address-cells = <1>;
+				#size-cells = <0>;
+				compatible = "lantiq,xrx200-mdio";
+			};
+
+			gphy-fw {
+				compatible = "lantiq,xrx200-gphy-fw", "lantiq,gphy-fw";
+				lantiq,rcu = <&rcu0>;
+				#address-cells = <1>;
+				#size-cells = <0>;
+
+				gphy0: gphy@20 {
+					reg = <0x20>;
+
+					resets = <&reset0 31 30>;
+					reset-names = "gphy";
+				};
+
+				gphy1: gphy@68 {
+					reg = <0x68>;
+
+					resets = <&reset0 29 28>;
+					reset-names = "gphy";
+				};
+			};
+                };
```

Comment:
Checkpatch found wrong indent in this line:
~~~
ERROR: code indent should use tabs where possible
#481: FILE: vr9.dtsi:481:
+                };$
~~~

Reply:
When DSA work, can I test on bt_homehub-v5a and ARV7519RW22 if you dont have this devices.
Im not a developer but I think can test and post results.

Reply:
> 
> 
> When DSA work, can I test on bt_homehub-v5a and ARV7519RW22 if you dont have this devices.
> Im not a developer but I think can test and post results.

You don't ask to test, this is open and public source/patches to source. You just give results, you are just generating noise, you should wait until @abajk has a patch for the dsa driver. Thank you for your intrest.

Reply:
excellent catch - thank you!
I just fixed this